target/scripts/startup/setup.d/mail_state.sh

```diff
@@ -105,10 +105,10 @@ function _setup_save_states() {
     # These two require the postdrop(103) group:
     chgrp -R postdrop "${STATEDIR}"/spool-postfix/{maildrop,public}
 
-    # After changing the group, special bits (set-gid, sticky) may be stripped, restore them:
-    # Ref: https://github.com/docker-mailserver/docker-mailserver/pull/3149#issuecomment-1454981309
-    chmod 1730 "${STATEDIR}/spool-postfix/maildrop"
-    chmod 2710 "${STATEDIR}/spool-postfix/public"
+    # These permissions rely on the `postdrop` binary (and alike) having proper SUID/SGID bits set.
+    # Ref: https://github.com/docker-mailserver/docker-mailserver/issues/3619
+    chmod 730 "${STATEDIR}/spool-postfix/maildrop"
+    chmod 710 "${STATEDIR}/spool-postfix/public"
```

Comment:
If maintainers think we should remove this altogether and just use the Debian defaults, I do not mind it either. But remember: these files are most likely in a persisted volume, hence, if we removed these lines, users would be stuck with the approach on `master`. Keeping these lines here is safer for now; we could add a TODO comment about removing this in future versions.

Reply:
> **Question:** What changed since the original #3149 PR to motivate this for you?:
> 
> * [You commented recently](https://github.com/docker-mailserver/docker-mailserver/pull/3149#issuecomment-1798683390) there that `allowPrivilegeEscalation: false` caused the issue for you.
>       
>   * But that PR itself fixed the same problem you encountered correct?

That's a good point, indeed.

>   * What is different now with the report in [docs: Kubernetes requies `allowPrivilegeEscalation: true` #3619](https://github.com/docker-mailserver/docker-mailserver/issues/3619) ? Why did the existing approach not work for that k8s user like it had for you?

I cannot answer this question TBH, because I don't know the answer 😆 I can only state that the setup with this PR makes way more sense. I do not know why I stopped seeing this error, I can only guess: at some point, I started using a completely custom Postfix configuration - maybe this is the culprit.

> > If maintainers think we should remove this altogether and just use the Debian defaults, I do not mind it either.
> > Keeping these lines here is safer for now; we could add a TODO comment about removing this in future versions.
> 
> You can't drop this to rely on Debian defaults.
> 
> They're not carried over into a fresh volume mount for these locations IIRC. That was the whole point we explicitly added them 😅

👍🏼 

> > ```shell
> > # These permissions rely on the `postdrop` binary (and alike) having proper SUID/SGID bits set.
> > ```
> 
> * Doesn't look like there is any SUID bit involved at all?
>       
>   * Only `setgid` / SGID on the `postdrop` binary.
>   * Possibly also `postqueue` (_it has SGID set too_), but I'm not sure about it's involvement with either dir.

Indeed, it's only SGID.

> * I agree that these extra special bits (_Sticky bit on `maildrop/` and SGID on `public/`_) don't seem to be necessary, at least for DMS (_dropping them will however diverge from Debian and be inconsistent vs no volume mount for state_).
> 
> 
> **NOTE:** The sticky-bit on `maildrop/` was likely historical from the [previous PR referenced Postfix release notes](https://github.com/vdukhovni/postfix/blob/1f2df46e5211065a16f78739308131122e185c7c/postfix/RELEASE_NOTES-1.1#L32-L34). Debian just may never have adjusted to that on their end, or it may be for supporting third-party software like `mailx` writing to `maildrop/`

I am pretty sure too that the sticky bit is not required :)